src/main/scala/viper/server/ViperServerRunner.scala

```diff
@@ -25,22 +28,27 @@ object ViperServerRunner {
     if (config.serverMode() == config.SERVER_MODE_LSP) {
       runLspServer(config)(executor)
     } else if (config.serverMode() == config.SERVER_MODE_HTTP) {
-      startHttpServer(config)(executor)
+      runHttpServer(config)(executor)
     } else {
       throw new RuntimeException(s"unknown server mode ${config.serverMode()}")
     }
   }
 
   /**
-    * Start VCS in HTTP mode.
+    * Run VCS in HTTP mode.
     */
-  def startHttpServer(config: ViperConfig)(executor: VerificationExecutionContext): Unit = {
+  def runHttpServer(config: ViperConfig)(executor: VerificationExecutionContext): Unit = {
     viperServerHttp = new ViperHttpServer(config)(executor)
     viperServerHttp.start()
+    // wait until server has been stopped:
+    Await.ready(viperServerHttp.stopped(), Duration.Inf)
```

Comment:
Are there any significant pros or cons of using Await rather than attaching a callback to the future's completion?

Reply:
I used `Await` here because this is basically the main function and imho I found it weird before that it basically immediately returns but keeps some futures running